packages/rest/src/lib/handlers/SequentialHandler.ts

```diff
@@ -329,17 +326,18 @@ export class SequentialHandler implements IHandler {
 					data: requestData,
 					retries,
 				},
-				res.clone(),
+				{ ...res },
```

Comment:
Does this do the same? Will this allow using the body twice?

Reply:
with `undici.request` the body is a ReadableStream so I'm not sure how it could be cloned to the same degree tbh

Reply:
I don't think this works, you're just copying a reference to the readable stream mat this point.

Because .body is readonly you also can't just pipe body to two passthroughs and replace body on the original response, though I suppose you could try reassigning res to { ...res, body: bodyStreamA } and use { ...res, body: bodyStreamB } as the emitted value (slowdowns are okay here, that's why we check for the listener)

Reply:
It doesn't work, but working with typescript and cloning streams doesn't work great :/. If you can work it out and send a PR to my fork I'd be more than happy to bring it in 😉

Reply:
uggh right, this isn't a *normal* readable, its undicis bodyreadable which isn't even exported and its not a web stream so can't just tee it. We'd have to reimplement undici's BodyReadable but as a BodyPassthrough in order to do this really

Reply:
We can use `stream/consumers` for the BodyMixin stuff, the main problem is getting typescript to like it

Reply:
I'll take a look at that later today. 

This whole thing is making me wonder if it's really worth it to use undicis proprietary request over the spec fetch implementation. Yes it's faster, but at some point the fetch api provides enough niceties that it's worth the difference (also should a discord bot ever be making that many req/s anyways? Unless you're proxying all requests from a sharded bot to a single instance). Just not sure if we're at that point or not

Reply:
This seems to be the only troublesome part, and `node-fetch`'s body was a ReadableStream as well (their cloning doesn't work great either). Using `request` also gives us a lower-leveled implementation that is much more customizable; for example, users of the library can now use proxies ( #3039 ), set custom request timeouts on header/body parsing, logging (using diagnostic channel), and much more. With fetch, you're limited to what the spec allows, and it isn't much...

Reply:
It sounds like we lose a good bit of that stuff if we clone it ourselves

Maybe undici would be open to adding a clone function to responsedata so that types make a whole lot more sense

Otherwise I'll look at how node-fetch did it (I think I have a pretty good guess) and see if we can apply it here

Reply:
Thinking it over... we can remove these events entirely, as undici provides diagnostic_channel support. Here's an example of getting the body sent to a server:

```js
import diagnosticsChannel from 'diagnostics_channel'
import { request } from 'undici';

diagnosticsChannel.channel('undici:request:create').subscribe(({ request }) => {
  console.log(request.body.toString()); // 'Hello, world!'
});

await request('http://httpbin.org/post', {
  method: 'POST',
  headers: {
    'hello': 'world'
  },
  body: 'Hello, world!'
})

```

Reply:
I've gone ahead and removed `request` and `response` events,
request - https://github.com/nodejs/undici/blob/main/docs/api/DiagnosticsChannel.md#undicirequestcreate
response - https://github.com/nodejs/undici/blob/main/docs/api/DiagnosticsChannel.md#undicirequesttrailers

Reply:
trailers does not satisfy the same functionality that our response event did. Undici only provides the trailers, not the returned body (because that would involve cloning it, an inherently slow operation. This event was only really useful for when there could potentially be something wrong with djs and we needed to see the raw api response body that was causing it.

Reply:
I think they'd accept a PR, considering that it would only be cloned if the channel is being subscribed to.

Reply:
The body is sent in the diagnostic channel, although not cloned & hidden via a symbol. Should theoretically be possible to clone and expose it in this event,

Reply:
yeah, took a look at the code, and the *very* indirect way to get the body for now (and it should be since its not cloned) and I don't mind it. This seems like something that undici should probably have anyways, so we shouldn't stall this PR on that basis.